keras_retinanet/preprocessing/df_generator.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+
+from keras_retinanet.preprocessing.csv_generator import CSVGenerator, Generator
+
+
+class DfGenerator(CSVGenerator):
+    """Custom generator intented to work with in-memory Pandas' dataframe."""
+    def __init__(self, df, class_mapping, cols, base_dir='', **kwargs):
+        """Initialization method.
+
+        Arguments:
+            df: Pandas DataFrame containing paths, labels, and bounding boxes.
+            class_mapping: Dict mapping label_str to id_int.
+            cols: Dict Mapping 'col_{filename/label/x1/y1/x2/y2} to corresponding df col.
```

Comment:
What does this do exactly? So the user has to input a dictionary here to map cols from their df to col_filename, col_label, etc. so that we can use those names in the generator?

Would it alternatively be okay to force the df to have cols named `filename, label, x1, y1, x2, y2` ? And if this is not the case, raise an exception.

Reply:
1. Yes, so we don't force the user to name its Pandas df columns as we want.
2. If you prefer to have a fixed column names, sure I can do that.

Reply:
I'm not sure. @de-vri-es what do you propose ?

Reply:
It could be a useful feature, but please drop the `col_` prefix from the keys. Since the argument is already called `cols` I think that's clear enough.

A default value which does an identity mapping probably make sense for the `cols` argument.